service/instance.py

```diff
@@ -2035,6 +2056,10 @@ def run_instance_action(user, identity, instance_id, action_type, action_params)
         action_type = 'suspend'
 
     logger.info("User %s has initiated instance action %s to be executed on Instance %s" % (user, action_type, instance_id))
+    if action_type in ('start', 'resume', 'unshelve'):
+        logger.info('Going to check the allocation of the instance due to the action type: %s', action_type)
```

Comment:
See comment above w.r.t to logging. This may be useful while you're developing the feature, but would argue shouldn't be part of the codebase.

Reply:
I prefer to use logging statements like this to help debugging issues. It shouldn't be a problem when you have a good log aggregator with flexible filtering rules.